website/1.0.1/src/site/markdown/tutorial_cloud_support.md

```diff
@@ -0,0 +1,228 @@
+
+`CLOUD_ENABLED` must be set to true if the user would like to use Helix cloud support. `CLOUD_PROVIDER` is the type of the cloud environment. Besides the few well-known public cloud providers, the user can also define his/her cloud environment as "CUSTOMIZED". `CLOUD_ID` is an optional field. It can be used to record any specific metadata information the user would like to record, e.g. the ID for the particular cluster inside a cloud environment, etc. If the user chooses to use the provider that already has default implementations in Helix, e.g., Azure, he does not need to provide the last two fields, as Helix already provides the default value for these two fields in system property, which is considered as a bundle with Azure implementation that would be discussed later. If the user uses customized providers, or chooses some other cloud environment that has not been implemented in Helix yet, the user needs to provide the last two fields and/or `CLOUD_ID` depending on his/her usage.  
+          
+| Field | Meaning |
+| ------------------------- | ------ |
+| CLOUD_ENABLED | determine whether the cluster is inside cloud environment and use Helix cloud support |
+| CLOUD_PROVIDER | denote what kind of cloud environment the cluster is in, e.g. Azure, AWS, GCP, or CUSTOMIZED |
+| CLOUD_ID | the specific id in cloud environment that belongs to this cluster |
+| CLOUD_INFO_SOURCE | the source for retrieving the cloud information. |
+| CLOUD_INFO_PROCESSOR_NAME | the name of the function that processes the fetching and parsing of cloud information |
+
+Users could use either REST API or Java API to set these configs.
+
+##### REST API Examples
+Helix enhanced current cluster creation REST API as well as Java API with extra fields that represent cloud related input. For example, in the following modified `createCluster` API, `addCloudConfig` is a Boolean value denotes whether to create with cloud config, and the `cloudConfigManifest` is the cloud config string, which will be converted to a Znode record.
+
+```
+@PUT
+  @Path("{clusterId}")
+  public Response createCluster(@PathParam("clusterId") String clusterId, @DefaultValue("false") @QueryParam("recreate") String recreate,
+      @DefaultValue("false") @QueryParam("addCloudConfig") String addCloudConfig, String cloudConfigManifest)
+```
+
+Besides the enhanced cluster creation API, Helix also provides a set of cloud specific APIs in Java and REST that handles the get/add/update of cloud config. 
+   
+* Create cluster and add meanwhile cloud config related information to ZK.
+
+```
+$ curl -X PUT -H "Content-Type: application/json" http://localhost:1234/admin/v2/clusters/myCluster?addCloudConfig=true -d '
+{
+    "simpleFields" : 
+    {
+        "CLOUD_ENABLED" : "true",
+        "CLOUD_PROVIDER": "AWS",
+        "CLOUD_ID" : "12345"
+        "CLOUD_INFO_SOURCE": {"http://169.254.169.254/"}
+        "CLOUD_INFO_PROCESSOR_NAME": "AWSCloudInformationProcesser"
```

Comment:
what is CLOUND_INFO_PROCESSOR_NAME used for in Helix?

Reply:
@kaisun2000 Helix implemented processor for Azure now, and the name is `AzureCloudInstanceInformationProcessor`